### PR TITLE
Run `git cherry-pick --abort` after it fails

### PR DESCRIPTION
**Pull request by komamitsu**
## Context

The auto-PR backport GHA workflow [failed](https://github.com/scalar-labs/scalardb/actions/runs/5407948409/jobs/9826502966) generated the unexpected large diffs and conflicts (https://github.com/scalar-labs/scalardb/pull/922 and https://github.com/scalar-labs/scalardb/pull/923). But as for branch `3.9`, it succeeds on my local environment with the expected diff. I think the first cherry-pick on branch `3.7` generated dirty changes and the following cherry-picks unexpectedly used it.

## Changes

This PR adds the following improvement to fix the problem.
- Execute `git cherry-pick --abort` when cherry-pick fails
- Cleanup the working dir just in case

### CODE REVIEW

File: ci/auto-pr/create_pull_requests [hunk 1]
```diff
@@ -39,6 +46,8 @@ function cherry_pick_and_create_pull_request () {
 }
 
 function create_issue () {
+  local branch=$1
```
Comment (komamitsu) [inline ci/auto-pr/create_pull_requests:49]: This is a bit unrelated bug fix that the global variable is used unexpectedly...

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
